esphome/components/sps30/sps30.cpp

```diff
@@ -0,0 +1,256 @@
+#include "sps30.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace sps30 {
+
+static const char *TAG = "sps30";
+
+static const uint16_t SPS30_CMD_GET_ARTICLE_CODE = 0xD025;
+static const uint16_t SPS30_CMD_GET_SERIAL_NUMBER = 0xD033;
+static const uint16_t SPS30_CMD_GET_FIRMWARE_VERSION = 0xD100;
+
+static const uint16_t SPS30_CMD_START_CONTINUOUS_MEASUREMENTS = 0x0010;
+static const uint16_t SPS30_CMD_START_CONTINUOUS_MEASUREMENTS_ARG = 0x0300;
+static const uint16_t SPS30_CMD_GET_DATA_READY_STATUS = 0x0202;
+static const uint16_t SPS30_CMD_READ_MEASUREMENT = 0x0300;
+static const uint16_t SPS30_CMD_STOP_MEASUREMENTS = 0x0104;
+static const uint16_t SPS30_CMD_SET_AUTOMATIC_CLEANING_INTERVAL_SECONDS = 0x8004;
+static const uint16_t SPS30_CMD_START_FAN_CLEANING = 0x5607;
+static const uint16_t SPS30_CMD_SOFT_RESET = 0xD304;
+static const size_t SERIAL_NUMBER_LENGTH = 8;
+
+void SPS30Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up sps30...");
+  this->write_command_(SPS30_CMD_SOFT_RESET);
+  delayMicroseconds(500000);
+  /// Firmware version identification
+  if (!this->write_command_(SPS30_CMD_GET_FIRMWARE_VERSION)) {
+    this->error_code_ = FIRMWARE_VERSION_REQUEST_FAILED;
+    this->mark_failed();
+    return;
+  }
+
+  uint16_t raw_firmware_version[4];
+  if (!this->read_data_(raw_firmware_version, 4)) {
+    this->error_code_ = FIRMWARE_VERSION_READ_FAILED;
+    this->mark_failed();
+    return;
+  }
+  ESP_LOGD(TAG, "  Firmware version v%0d.%02d", (uint16_t(raw_firmware_version[0]) >> 8),
```

Comment:
```suggestion
  ESP_LOGD(TAG, "  Firmware version v%0d.%02d", (raw_firmware_version[0] >> 8),
```

The value is already an `uint16_t`

Reply:
Fixed :)